libs/utils/src/lib/services/broadcast-channel/broadcast-channel.service.ts

```diff
@@ -0,0 +1,95 @@
+import { isPlatformBrowser } from '@angular/common';
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { EMPTY, fromEvent, Observable } from 'rxjs';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class BroadcastChannelService {
+	private broadcastChannel: Record<string, BroadcastChannel> = {};
+
+	constructor(@Inject(PLATFORM_ID) private platformId: string) {}
+
+	/**
+	 * initChannel
+	 *
+	 * The initChannel method initializes a new BroadcastChannel instance.
+	 *
+	 * @param channelName{string} - The name of the Broadcast Channel.
+	 */
+	public initChannel(channelName: string): void {
+		if (!channelName) {
+			console.warn('channelName is required');
+
+			return;
+		}
+
+		if (isPlatformBrowser(this.platformId) && !this.broadcastChannel[channelName]) {
+			this.broadcastChannel[channelName] = new BroadcastChannel(channelName);
+		}
+	}
+
+	/**
+	 * closeChannel
+	 *
+	 * The closeChannel method closes a selected BroadcastChannel instance.
+	 *
+	 * @param channelName{string} - The name of the Broadcast Channel.
+	 */
+	public closeChannel(channelName: string): void {
+		if (!channelName) {
+			console.warn('channelName is required');
+
+			return;
+		}
+
+		if (!this.broadcastChannel[channelName]) {
+			return;
+		}
+
+		this.broadcastChannel[channelName].close();
+		delete this.broadcastChannel[channelName];
+	}
+
+	/**
+	 * postMessage
+	 *
+	 * The postMessage method sends a message to a selected BroadcastChannel instance.
+	 *
+	 * @param channelName{string} - The name of the Broadcast Channel.
+	 * @param message{any} - The payload to send through the channel.
+	 */
+	public postMessage(channelName: string, message: any): void {
+		if (!channelName) {
```

Comment:
Because you already do this check in the `initChannel` method, this basically already gets handled by the next if check, as it can never be created to begin with.

Reply:
The `postMessage` can be called without calling the `initChannel` first. This was my train of tought when repeating these checks. As said in a previous comment, I'll simplify the checks and messages.